Library/Homebrew/extend/os/linux/compilers.rb

```diff
@@ -4,8 +4,8 @@
 class CompilerSelector
   sig { returns(String) }
   def self.preferred_gcc
-    # gcc-5 is the lowest gcc version we support on Linux.
-    # gcc-5 is the default gcc in Ubuntu 16.04 (used for our CI)
-    "gcc@5"
+    # gcc-11 is the lowest gcc version we support on Linux.
+    # gcc-11 is the default gcc in Ubuntu 22.04 (used for our CI)
+    "gcc@11"
```

Comment:
Can we just get rid of this and use the latest GCC?

I think this is part of our discussion here: https://github.com/Homebrew/brew/pull/13577#discussion_r946494611

Reply:
Note that this would mean brewed GCC would always be installed on CI as well since it would see GCC 11 as too old.  Note sure if this is what you're arguing or not - the prior discussion only mentions runtime libraries.

(If not, further modifications could however be made so host GCC 11 is used if available, but latest GCC is installed for runtime if older than 11.)

Reply:
I feel like:
- ideally we'd use the system `glibc` and `gcc` on our primary CI builds
- users end up using the same `glibc` and `gcc` versions from the host or from us if their host version is newer or older

Reply:
> (If not, further modifications could however be made so host GCC 11 is used if available, but latest GCC is installed if older than 11.)

Yes, this is what I'm looking for. Apologies for the lack of clarity there.

Reply:
> > (If not, further modifications could however be made so host GCC 11 is used if available, but latest GCC is installed if older than 11.)
> 
> Yes, this is what I'm looking for. Apologies for the lack of clarity there.

IMO we'll be in a better place if we just use the same GCC major version all the time.

Reply:
I agree that using the same GCC major version is simpler. See https://github.com/Homebrew/brew/pull/13577#discussion_r951632245